readthedocs/rtd_tests/tests/projects/test_version_sorting.py

```diff
@@ -65,3 +71,95 @@ def test_sort_alpha(self):
             ['latest', 'carrot', 'banana', 'apple'],
             [v.slug for v in sort_version_aware(versions)],
         )
+
+    def test_sort_git_master(self):
+        identifiers = ['master', '1.0', '2.0', '1.1', '1.9', '1.10']
+        self.project.repo_type = REPO_TYPE_GIT
+        self.project.save()
+        self.project.versions.get(slug=LATEST).delete()
+
+        for identifier in identifiers:
+            get(
+                Version,
+                project=self.project,
+                type=BRANCH,
+                identifier=identifier,
+                verbose_name=identifier,
+                slug=identifier,
+            )
+
+        versions = list(Version.objects.filter(project=self.project))
+        self.assertEqual(
+            ['master', '2.0', '1.10', '1.9', '1.1', '1.0'],
+            [v.slug for v in sort_version_aware(versions)],
+        )
+
+    def test_sort_hg_default(self):
+        identifiers = ['default', '1.0', '2.0', '1.1', '1.9', '1.10']
+        self.project.repo_type = REPO_TYPE_HG
+        self.project.save()
+        self.project.versions.get(slug=LATEST).delete()
+
+        for identifier in identifiers:
+            get(
+                Version,
+                project=self.project,
+                type=BRANCH,
+                identifier=identifier,
+                verbose_name=identifier,
+                slug=identifier,
+            )
+
+        versions = list(Version.objects.filter(project=self.project))
+        self.assertEqual(
+            ['default', '2.0', '1.10', '1.9', '1.1', '1.0'],
+            [v.slug for v in sort_version_aware(versions)],
+        )
+
+    def test_sort_bzr_latest(self):
+        """
+        BZR doesn't have a name for "master",
+        so here master gets sorted by its ascii value.
+        """
+        identifiers = ['master', '1.0', '2.0', '1.1', '1.9', '1.10']
+        self.project.repo_type = REPO_TYPE_BZR
+        self.project.save()
+        self.project.versions.get(slug=LATEST).delete()
+
+        for identifier in identifiers:
+            get(
+                Version,
+                project=self.project,
+                type=BRANCH,
+                identifier=identifier,
+                verbose_name=identifier,
+                slug=identifier,
+            )
+
+        versions = list(Version.objects.filter(project=self.project))
+        self.assertEqual(
+            ['2.0', '1.10', '1.9', '1.1', '1.0', 'master'],
+            [v.slug for v in sort_version_aware(versions)],
+        )
+
+    def test_sort_svn_trunk(self):
+        identifiers = ['/trunk/', '1.0', '2.0', '1.1', '1.9', '1.10']
```

Comment:
not really sure about this one, but it's as fallback_branch, so :man_shrugging:

and not like we have many of these :eyes: